Rover/GCS_MAVLink_Rover.cpp

```diff
@@ -228,9 +228,9 @@ void GCS_MAVLINK_Rover::send_water_depth()
             continue;
         }
 
-        // get temperature
+        // get temperature (with external sensor override support)
         float temp_C;
-        if (!s->get_temp(temp_C)) {
+        if (!rangefinder->get_temp_with_override(ROTATION_PITCH_270, temp_C)) {
```

Comment:
This code should not change. Please look at the other uses of the temp _SRC in other libraries. Look for where AP_TEMPERATURE_SENSOR_ENABLED is used in AP_ESC_Telem and AP_BattMonintor. From the vehicle's perspective it should not know/care where the temperature is coming from, the temp library overrides the rangefinder's value.

If we did it this way, all vehicles would have to check which is not very good abstraction.